### PR TITLE
Make `BaseGroovyExtension` and `BaseKotlinExtension` public

### LINKED ISSUE #1911: Regression in gradle/6.23.0. Unable to configre ktlint
## summary of problem
```kt
spotless {
    kotlin { 
        ktlint()
    }
}
```
Having the simple block below highlights `ktlint()`. Hovering on it shows the message "Type BaseKotlinExtension.KtlintConfig! is inaccessible in this context due to: public open inner class KtlintConfig defined in com.diffplug.gradle.spotless.BaseKotlinExtension". But spotless works fine. Now if i try to access any method of ktlint config like below red squiggly line appers below that method and hovering over them shows the message "Cannot access 'BaseKotlinExtension': it is package-private in 'com.diffplug.gradle.spotless'"
```kt
spotless {
    kotlin {
        ktlint()
            .customRuleSets(listOf("io.nlopez.compose.rules:ktlint:0.3.3"))
    }
}
```
## Gradle or Maven version
Gradle 8.4
## spotless version
gradle/6.23.0
## operating system and version
Windows 11
## copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
See above
## copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`
```
> Configure project :app
w: file:///D:/[...]/android/app/build.gradle.kts:46:9: Type BaseKotlinExtension.KtlintConfig! is inaccessible in this context due to: public open inner class KtlintConfig defined in com.diffplug.gradle.spotless.BaseKotlinExtension
w: file:///D:/[...]/android/app/build.gradle.kts:46:9: Type BaseKotlinExtension.KtlintConfig! is inaccessible in this context due to: public open inner class KtlintConfig defined in com.diffplug.gradle.spotless.BaseKotlinExtension
e: file:///D:/[...]/android/app/build.gradle.kts:47:14: Cannot access 'BaseKotlinExtension': it is package-private in 'com.diffplug.gradle.spotless'

FAILURE: Build failed with an exception.

* Where:
Build file 'D:\[...]\android\app\build.gradle.kts' line: 47

* What went wrong:
Script compilation error:

  Line 47:             .customRuleSets(listOf("io.nlopez.compose.rules:ktlint:0.3.3"))
                        ^ Cannot access 'BaseKotlinExtension': it is package-private in 'com.diffplug.gradle.spotless'
```                        

If you're just submitting a feature request or question, no need for the above.

### PR DESCRIPTION
**Pull request by Goooler**
- Follow up https://github.com/diffplug/spotless/pull/1890#issuecomment-1827263031.
- Closes #1911.

### CODE REVIEW

**Comment by msfjarvis** [comment]
In the mean time I've worked around it like this: https://github.com/msfjarvis/compose-lobsters/commit/33425f16a10e

File: plugin-gradle/CHANGES.md [hunk 1]
```diff
@@ -3,10 +3,12 @@
 We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (starting after version `3.27.0`).
 
 ## [Unreleased]
+### Fixed
+* Make `BaseGroovyExtension` and `BaseKotlinExtension` public. ([#1912](https://github.com/diffplug/spotless/pull/1912))
 
 ## [6.23.0] - 2023-11-27
```
Comment (TWiStErRob) [inline plugin-gradle/CHANGES.md:9]: also mention breaking change for previous release?

**Review by TWiStErRob — APPROVED** [review]

File: plugin-gradle/CHANGES.md (hunk 1 above)
Comment (Goooler) [reply plugin-gradle/CHANGES.md:9]: That's a bug instead of intended behavior.

Comment (Goooler) [reply plugin-gradle/CHANGES.md:9]: ba92c77add9ee3f5e0484e6caa37486893629a34

**Comment by nedtwigg** [comment]
Fixed in `plugin-gradle 6.23.1`. It has been published in the Gradle plugin repository, but the maven central sync appears to be jammed for now. Should resolve in a few hours.

**Pull request merged**